cmd/podman/ps.go

```diff
@@ -250,106 +255,120 @@ func psCmd(c *cliconfig.PsValues) error {
 		}
 	}
 
-	if !opts.Latest {
-		// Get all containers
-		containers, err := runtime.GetContainers(filterFuncs...)
-		if err != nil {
-			return err
-		}
-
-		// We only want the last few containers
-		if opts.Last > 0 && opts.Last <= len(containers) {
-			return errors.Errorf("--last not yet supported")
-		} else {
-			outputContainers = containers
-		}
-	} else {
-		// Get just the latest container
-		// Ignore filters
-		latestCtr, err := runtime.GetLatestContainer()
-		if err != nil {
-			return err
-		}
-
-		outputContainers = []*libpod.Container{latestCtr}
-	}
-
 	maxWorkers := shared.Parallelize("ps")
 	if c.GlobalIsSet("max-workers") {
 		maxWorkers = c.GlobalFlags.MaxWorks
 	}
 	logrus.Debugf("Setting maximum workers to %d", maxWorkers)
 
-	pss := shared.PBatch(outputContainers, maxWorkers, opts)
-	if opts.Sort != "" {
-		pss, err = sortPsOutput(opts.Sort, pss)
-		if err != nil {
-			return err
+	for {
+		if watch {
```

Comment:
Can we move the contents of this loop into a helper for readability, and so we don't have to unconditionally call them in an infinite loop?

Reply:
are you wanting something like:
```
if !watch {
  helperFunc()
} else {
   for {
     helperfunc()
   }
}

Reply:
Yes